### ISSUE TITLE
Training process crashed 

### ISSUE THREAD

**Issue by libeineu** [pytorch/fairseq#1018]
Hello, when I used the Fairseq-0.6.2 to train a ZH-EN transformer model, the training process crashed after 3 epochs, which the GPUS were fully located and the log was stuck. 
![image](https://user-images.githubusercontent.com/26756710/62996442-4e080b80-be97-11e9-88d5-e8cc14ad10f4.png)
![image](https://user-images.githubusercontent.com/26756710/62996453-58c2a080-be97-11e9-9b6b-8edfe9efe8bc.png)
I have filtered the longest sentence by 120, but it can only successfully trained 2 epochs and crashed at the same step.

**Comment by libeineu** [comment]
Thanks for your advice! I have used the no_c10d mode. I used Fairseq-0.6.2 and pytorch 1.1.0 ,CUDA10. I have met the same problem when I trained other models before, and I filtered the sentences whose length are longer than 200, but it sometime crashed. The detail model hyper-parameter is as the following:
adam_betas                               = (0.9, 0.997)
adam_eps                                 = 1e-08
adaptive_input                           = False
adaptive_softmax_cutoff                  = None
adaptive_softmax_dropout                 = 0
arch                                     = transformer_t2t_wmt_en_de_big
attention_dropout                        = 0.1
bucket_cap_mb                            = 25
clip_norm                                = 0.0
cpu                                      = False
criterion                                = label_smoothed_cross_entropy
curriculum                               = 0
data                                     = ['data-bin/wmt-en2de']
ddp_backend                              = no_c10d
decoder_attention_heads                  = 16
decoder_embed_dim                        = 1024
decoder_embed_path                       = None
decoder_ffn_embed_dim                    = 4096
decoder_input_dim                        = 1024
decoder_layers                           = 6
decoder_learned_pos                      = False
decoder_normalize_before                 = True
decoder_output_dim                       = 1024
device_id                                = 0
distributed_backend                      = nccl
distributed_init_method                  = None
distributed_port                         = -1
distributed_rank                         = 0
distributed_world_size                   = 8
dropout                                  = 0.1
encoder_attention_heads                  = 16
encoder_embed_dim                        = 1024
encoder_embed_path                       = None
encoder_ffn_embed_dim                    = 4096
encoder_layers                           = 6
encoder_learned_pos                      = False
encoder_normalize_before                 = True
fix_batches_to_gpus                      = False
fp16                                     = True
fp16_init_scale                          = 128
fp16_scale_tolerance                     = 0.0
fp16_scale_window                        = None
k_only                                   = True
keep_interval_updates                    = -1
keep_last_epochs                         = 10
label_smoothing                          = 0.1
lazy_load                                = False
left_pad_source                          = True
left_pad_target                          = False
log_format                               = None
log_interval                             = 100
lr                                       = [0.0007]
lr_scheduler                             = inverse_sqrt
lr_shrink                                = 0.1
max_epoch                                = 21
max_relative_length                      = 8
max_sentences                            = None
max_sentences_valid                      = None
max_source_positions                     = 1024
max_target_positions                     = 1024
max_tokens                               = 2048
max_update                               = 0
memory_efficient_fp16                    = False
min_loss_scale                           = 0.0001
min_lr                                   = 1e-09
momentum                                 = 0.99
no_epoch_checkpoints                     = False
no_progress_bar                          = True
no_save                                  = False
no_token_positional_embeddings           = False
num_workers                              = 0
optimizer                                = adam
optimizer_overrides                      = {}
raw_text                                 = False
relu_dropout                             = 0.1
required_batch_size_multiple             = 8
reset_lr_scheduler                       = False
reset_optimizer                          = False
restore_file                             = checkpoint_last.pt
save_dir                                 = checkpoints/wmt-en2de/big-rpr
save_interval                            = 1
save_interval_updates                    = 0
seed                                     = 1
sentence_avg                             = False
share_all_embeddings                     = True
share_decoder_input_output_embed         = False
skip_invalid_size_inputs_valid_test      = False
source_lang                              = en
target_lang                              = de
task                                     = translation
tensorboard_logdir                       = checkpoints/wmt-en2de/big-rpr
threshold_loss_scale                     = None
train_subset                             = train
update_freq                              = [4]
upsample_primary                         = 1
user_dir                                 = None
valid_subset                             = valid
validate_interval                        = 1
warmup_init_lr                           = 1e-07
warmup_updates                           = 8000
weight_decay                             = 0.0

发送自 Windows 10 版邮件<https://go.microsoft.com/fwlink/?LinkId=550986>应用

________________________________
发件人: Myle Ott <notifications@github.com>
发送时间: Monday, August 19, 2019 10:07:02 PM
收件人: pytorch/fairseq <fairseq@noreply.github.com>
抄送: LiBei <libei_neu@outlook.com>; Author <author@noreply.github.com>
主题: Re: [pytorch/fairseq] Training process crashed (#1018)


Can you clarify exactly the training command you ran, dataset, PyTorch version, etc?

This is usually caused by a failure on one of the workers, which causes the other works to hang. Please also try rerunning with --ddp-backend=no_c10d, which is usually a bit more robust.

―

**Comment by lematt1991** [comment]
Closing due to inactivity.  Please open a new issue if you are still experiencing the issue.
